api/pkg/apis/v1alpha1/utils/utils.go

```diff
@@ -234,8 +227,14 @@ func FormatObject(obj interface{}, isArray bool, path string, format string) ([]
 				jData, _ = json.Marshal(ob)
 			}
 		}
+	} else {
+		if format == "yaml" {
+			jData, err = yaml.Marshal(obj)
+		} else {
+			jData, err = json.Marshal(obj)
```

Comment:
We should keep the same way to report error. Why we omit the error in other places but keep it in this place?

Reply:
In the original version, we omit all errors. Should we keep it or record as many error as possible?

Reply:
I prefer we always return error except we think the caller doesn't need to know it.